### PR TITLE
make lesson suitable for teaching with locally-built (serverless) pages

### PR DESCRIPTION
**Pull request by tobyhodges**
See discussion here for more background: https://github.com/datacarpentry/datacarpentry.github.io/issues/542

In short, some Carpentries lessons use `/guide/` for their Instructor Notes page and others use `/guide/index.html` (the default defined in `_config.yml`). This PR removes permalinks with a trailing slash from the YAML front matter of the Instructor Notes (`_extras/guide.md`) and other Extras files, consistent with new lessons created with [the lesson template](https://github.com/carpentries/styles/). Although there's no functional difference in the online versions of the lesson pages, pages with a trailing slash in the permalink will result in **broken links in the version of the lesson built locally**. We'd like local builds to be usable e.g. in workshops taking place at locations with limited or unreliable Internet access. 

Keeping the paths to these files consistent will also help us avoid broken links on the [Software Carpentry Lessons page](https://software-carpentry.org/lessons/), and ensure that equivalent paths in new lessons created with the template are consistent with previously-developed lessons like this one.

I did a sweep through the lesson and adjusted internal links as part of this PR. If and when this is merged, I'll also make sure the link on https://software-carpentry.org/lessons/ to the Instructor Notes page idn't broken, and make another sweep through the lesson pages too.

### CODE REVIEW

**Review by maxim-belkin — APPROVED** [review]

**Comment by maxim-belkin** [comment]
Thanks, Toby!

**Comment by maxim-belkin** [comment]
@swcarpentry/python-novice-inflammation-maintainers, once you review and approve it, make sure to "Rebase and merge" instead of "Squash and merge.

**Review by ldko — APPROVED** [review]
This looks good to me. Thank you, Toby! :+1:

**Pull request merged**
